### PR TITLE
refactor: rely on nextclade output for sequence diagnostic

### PR DESCRIPTION
**Pull request by rneher**
This PR uses the newly available info in metadata to simplify and speed up the filtering for by clock deviation and SNP clusters

The threshold can be set as arguments (only in script yet), not the config.

supersedes #581

### CODE REVIEW

**Comment by rneher** [comment]
@huddlej after https://github.com/nextstrain/ncov-ingest/pull/185 is merged, this one should be able to go in without trouble. Covariants isn't using the diagnostic file anymore.

**Comment by rneher** [comment]
@huddlej have a look:

https://nextstrain.org/staging/trial/no_diagnostic/ncov/africa?l=clock

(and analoga)
 
this should be an easy win for making the workflow simpler.

**Review by huddlej — APPROVED** [review]
This is excellent! I had a question about the switch from `read_metadata` (which supports more generic delimiters but has other issues like large memory use) to `pd.read_csv`, but this worked for me and makes a lot of sense.

We could use [the exact pandas code here from Augur](https://github.com/nextstrain/augur/blob/eb1e8502373a8fabbcd08d22d0c6e16fe2b4dcd3/augur/util_support/metadata_file.py#L91-L97) to make sure people with non-standard delimiters we already support are still supported:

```python
return pandas.read_csv(
            self.fname,
            sep=None,  # csv.Sniffer will automatically detect sep
            engine="python",
            skipinitialspace=True,
            dtype={"strain":"string", "name":"string"}
        ).fillna("")
```

File: scripts/diagnostic.py [hunk 1]
```diff
@@ -1,143 +1,43 @@
...
-    parser.add_argument("--output-flagged", type=str, required=True, help="Output of sequences flagged for exclusion with specific reasons")
+    parser.add_argument("--clock-filter", type=float, default=20, help="maximal allowed deviation from the molecular clock")
+    parser.add_argument("--snp-clusters", type=int, default=1, help="maximal allowed SNP clusters (as defined by nextclade)")
     parser.add_argument("--output-exclusion-list", type=str, required=True, help="Output to-be-reviewed addition to exclude.txt")
     args = parser.parse_args()
 
-    # load entire alignment and the alignment of focal sequences (upper case -- probably not necessary)
-    ref  = SeqIO.read(args.reference, 'genbank').seq
-    metadata, _ = read_metadata(args.metadata)
+    metadata = pd.read_csv(args.metadata, sep='\t')
```
[the remark this answers is not in the document]
Comment (rneher) [reply scripts/diagnostic.py:26]: the main reason here was that a data frame was more useful than a dict (and I like the "what is says on the tin" feel). I think without the sniffer, it uses by default a C++ engine which I guess is faster and possibly uses less memory.

[the remark this answers is not in the document]
Comment (rneher) [reply scripts/diagnostic.py:26]: but we can definitely use the snippet from augur.

**Comment by rneher** [comment]
We should probably make those 

https://github.com/nextstrain/ncov/pull/667/files#diff-a23ae4bd3f5f8c492703b229cae5da2bd5ec7c320ed5fc67f5240128916f320cR21-R22

parameters in the `snakemake` rule.

File: scripts/diagnostic.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (huddlej) [reply scripts/diagnostic.py:26]: Yeah, I wish we hadn't agreed to support the less standard syntax...there's still time in the future to change our minds though! :D

**Pull request merged**